tests/integration/src/setup.rs

```diff
@@ -0,0 +1,133 @@
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+//! Runtime integration tests setup & imports.
+
+pub use frame_support::{
+    assert_ok,
+    traits::{OnFinalize, OnInitialize},
+};
+pub use sp_core::H160;
+pub use sp_runtime::{AccountId32, MultiAddress};
+
+#[cfg(feature = "shibuya")]
+pub use shibuya::*;
+#[cfg(feature = "shibuya")]
+mod shibuya {
+    pub use shibuya_runtime::*;
+
+    /// 1 SBY.
+    pub const UNIT: Balance = SBY;
+}
+
+#[cfg(feature = "shiden")]
+pub use shiden::*;
+#[cfg(feature = "shiden")]
+mod shiden {
+    pub use shiden_runtime::*;
+
+    /// 1 SDN.
+    pub const UNIT: Balance = SDN;
+}
+
+#[cfg(feature = "astar")]
+pub use astar::*;
+#[cfg(feature = "astar")]
+mod astar {
+    pub use astar_runtime::*;
+
+    /// 1 ASTR.
+    pub const UNIT: Balance = ASTR;
+}
+
+pub const ALICE: AccountId32 = AccountId32::new([1_u8; 32]);
+pub const BOB: AccountId32 = AccountId32::new([2_u8; 32]);
+pub const CAT: AccountId32 = AccountId32::new([3_u8; 32]);
+
+pub const INITIAL_AMOUNT: u128 = 100_000 * UNIT;
+
+pub type SystemError = frame_system::Error<Runtime>;
+pub use pallet_balances::Call as BalancesCall;
+pub use pallet_dapps_staking as DappStakingCall;
+pub use pallet_proxy::Event as ProxyEvent;
+pub use pallet_utility::{Call as UtilityCall, Event as UtilityEvent};
+
+pub struct ExtBuilder {
+    balances: Vec<(AccountId32, Balance)>,
+}
+
+impl Default for ExtBuilder {
+    fn default() -> Self {
+        Self { balances: vec![] }
+    }
+}
+
+impl ExtBuilder {
+    pub fn balances(mut self, balances: Vec<(AccountId32, Balance)>) -> Self {
+        self.balances = balances;
+        self
+    }
+
+    pub fn build(self) -> sp_io::TestExternalities {
+        let mut t = frame_system::GenesisConfig::default()
+            .build_storage::<Runtime>()
+            .unwrap();
+
+        pallet_balances::GenesisConfig::<Runtime> {
+            balances: self.balances,
+        }
+        .assimilate_storage(&mut t)
+        .unwrap();
+
+        let mut ext = sp_io::TestExternalities::new(t);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+}
+
+pub fn new_test_ext() -> sp_io::TestExternalities {
+    ExtBuilder::default()
+        .balances(vec![
+            (ALICE, INITIAL_AMOUNT),
+            (BOB, INITIAL_AMOUNT),
+            (CAT, INITIAL_AMOUNT),
+        ])
+        .build()
+}
+
+pub fn run_to_block(n: u32) {
+    while System::block_number() < n {
+        DappsStaking::on_finalize(System::block_number());
+        System::set_block_number(System::block_number() + 1);
+        DappsStaking::on_initialize(System::block_number());
```

Comment:
Can we run the hooks for all pallets?
AFAIK, they cannot be just fetched and iterated over, while being able to call hooks on them (hope I'm wrong though!).

Reply:
Not a problem. I was originally thinking to add them as needed. After some second thoughts, I agree with you, will do so that we don't need to worry about adding more pallets overtime.

Reply:
Great! Now that I rethink what I wrote, there has to be a way to do it since it's done when network is live 😅

Reply:
Updated. I checked all pallets manually and added hooks if there are defined. `ParachainSystem` was skipped as it requires extra mocks but not required for tests. The pallets could be iterated via some tricky way, like macro or the runtime enum definition, but IMO it would be a bit overkill and edge cases would make more complicated.